snuba/query/expressions.py

```diff
@@ -0,0 +1,183 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Callable, Iterable, Iterator, Optional, Sequence
+
+from snuba.query.collections import NodeContainer
+
+
+class Expression(ABC):
```

Comment:
Not specific to this line, but I think this is missing literal subtypes of `Expression` for string, number, array, etc.

Reply:
right.